packages/crayons-core/src/components/data-table/data-table.tsx

```diff
@@ -634,9 +1007,9 @@ export class DataTable {
    * @returns {JSX.Element} table header row
    */
   renderTableHeader() {
-    const selectAllChecked = this.rows
-      .map((row) => row.id)
-      .every((rowId) => this.selected.includes(rowId));
+    const selectAllChecked = this.rows.every((row) =>
```

Comment:
The above operation can be a costly one if the no of rows and selected rows are huge in no. 

can we not just compare the length `this.rows.length === this.selected` instead ?.

Reply:
As we discussed, this is necessary to maintain selected state between pagination.